### ISSUE TITLE
Inductor can not fuse cat with a followed pointwise

### ISSUE THREAD

**Issue by shunting314** [pytorch/pytorch#125075]
### 🐛 Describe the bug

Check this example

```
    def test_concat_and_downcast(self):
        M = 30522
        N = 768
        PAD = 6

        @torch.compile
        def f(x):
            z = torch.cat([x, torch.zeros([PAD, N])], dim=0)
            return z, x.to(torch.float16)

        x = torch.randn(M, N)
        f(x)
```
The cat and the downcast can not be fused right now. But in principal we should be able to fuse them and save one whole load of tensor 'x'.

### Versions

.

**Comment by Chillee** [comment]
We'd fuse it with the pointwise cat codegen?

**Comment by shunting314** [comment]
> We'd fuse it with the pointwise cat codegen?

That requires the awareness of the other pointwise node that can be fused when we lower cat as a piontwise? I think it's a bit hard since the other pointwise node that can be fused can happen somewhere else.

Or an alternative is to do this in scheduler since the scheduler does global search and can try to fuse the pointwise node with the cat node. But the tricky thing here is these 2 nodes has different numels.

**Comment by IvanKobzarev** [comment]
Hi @shunting314 ,  doing issue scrapping, is this still a relevant issue or it was fixed?

**Comment by Chillee** [comment]
@IvanKobzarev from testing doesn't seem to be fixed

**Comment by eellison** [comment]
@shunting314 , do you remember the original context? mm padding ?

**Comment by shunting314** [comment]
> do you remember the original context? mm padding ?

I can not recall any more. maybe something I see from the kernels captured for Huggingface benchmark suite

**Comment by masnesral** [comment]
[old issue scrubbing] Looks like no changes; still repros.

**Comment by jansel** [comment]
I think we have pointwise cat config option that might do this now.

**Comment by masnesral** [comment]
> I think we have pointwise cat config option that might do this now. 

@jansel, could it be `force_pointwise_cat`? If so, it doesn't seem to affect the codegen for this repro at least.

**Comment by jansel** [comment]
Yes, that is what I was talking about.

**Comment by amritahs-ibm** [comment]
@shunting314 Is this issue fixed? or it still repros?

**Comment by jansel** [comment]
My agent was unable to reproduce this on da74fecc24c85f9694061e961858303c44be4338. Reopen if you see this on the latest PyTorch nightly
